Recipes/Controller/MainViewController.swift

```diff
@@ -0,0 +1,81 @@
+//
+//  MainViewController.swift
+//  Recipes
+//
+//  Created by krikaz on 3/12/20.
+//  Copyright © 2020 Lambda Inc. All rights reserved.
+//
+
+import UIKit
+
+class MainViewController: UIViewController {
+    
+    let networkClient = RecipesNetworkClient()
+    
+    var allRecipes: [Recipe] = [] {
+        didSet {
+            filterRecipes()
+        }
+    }
+    
+    var recipesTableViewController: RecipesTableViewController? {
+        didSet {
+            recipesTableViewController?.recipes = filteredRecipes
+        }
+    }
+    
+    var filteredRecipes: [Recipe] = [] {
+        didSet {
+            recipesTableViewController?.recipes = filteredRecipes
+        }
+    }
```

Comment:
Properties and methods should be private since we want to restrict access to them. If you let it like, it is `internal` by default (accessible from the module level which means the entire app here).

More : https://docs.swift.org/swift-book/LanguageGuide/AccessControl.html